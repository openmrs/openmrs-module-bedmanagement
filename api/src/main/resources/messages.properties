${project.parent.artifactId}.title=bedmanagement Module
${project.parent.artifactId}.admissionLocations=Admission Locations
${project.parent.artifactId}.bedTypes=Bed Types
${project.parent.artifactId}.bedTags=Bed Tags