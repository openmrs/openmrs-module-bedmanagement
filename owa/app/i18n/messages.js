export default {
    en: {
        ADMISSION_LOCATIONS: 'Admission Locations',
        BED_TYPES: 'Bed Types',
        BED_TAGS: 'Bed Tags',
        LOGOUT: 'Logout',
        CHOOSE_LANGUAGE: 'Choose language',
        ROW_REQUIRED_MSG: 'Row is required field',
        COLUMN_REQUIRED_MSG: 'Column is required field',
        ROW_SHOULD_GREATER_THAN_ZERO: 'Row value should be greater than 0',
        ROW_SHOULD_GREATER_THAN_LAYOUT_ROW: 'Row value should not be greater than layout row size',
        COLUMN_SHOULD_GREATER_THAN_ZERO: 'Column value should be greater than 0 and less than 11',
        COLUMN_SHOULD_GREATER_THAN_LAYOUT_COLUMN: 'Row value should not be greater than layout row size',
        FIX_ERROR_MSG: 'Fix error before submit',
        ADMISSION_LOCATION_SAVE_MSG: 'Admission location save successfully',
        BED_LAYOUT_SAVE_MSG: 'Admission location bed layout save successfully',
        BED_TAG_SAVE_MSG: 'Bed Tag save successfully',
        BED_TYPE_SAVE_MSG: 'Bed Type save successfully',
        BED_SAVE_MSG: 'Bed save successfully',
        DELETE_ADMISSION_LOCATION_CONFIRM_MESSAGE:
            'Are you sure you want to delete admission location {location_name}?',
        DELETE_BED_CONFIRM_MESSAGE: 'Are you sure you want to delete bed number {bed_number}?',
        DELETE_BED_TAG_CONFIRM_MSG: 'Are you sure you want to delete bed Tag {bed_tag_name}?',
        DELETE_BED_TYPE_CONFIRM_MSG: 'Are you sure you want to delete bed type {bed_type_name}?',
        DELETE_SUCCESSFULLY: 'Delete successfully',
        SUCCESS: 'Success',
        ERROR: 'Error',
        INFO: 'Info',
        SAVE: 'Save',
        SAVING: 'Saving...',
        CANCEL: 'Cancel',
        ADD: 'Add',
        EDIT: 'Edit',
        DELETE: 'Delete',
        ADD_BED: 'Add Bed',
        ROWS: 'Rows',
        ROW: 'Row',
        COLUMNS: 'Columns',
        COLUMN: 'Column',
        SET_LAYOUT: 'Set Layout',
        LOCATION: 'Location',
        BED: 'Bed',
        BED_NUMBER: 'Bed Number',
        BED_TYPE: 'Bed Type',
        WARD: 'Ward',
        PARENT_LOCATION: 'Parent Location',
        NONE: 'None',
        NAME: 'Name',
        DESCRIPTION: 'Description',
        DISPLAY_NAME: 'Display Name',
        ACTION: 'Action',
        ADD_NEW: 'Add New',
        BED_TAG: 'Bed Tag',
        EXISTING_BED_TYPES: 'Existing Bed Types',
        EXISTING_BED_TAGS: 'Existing Bed Tags'
    }
};
