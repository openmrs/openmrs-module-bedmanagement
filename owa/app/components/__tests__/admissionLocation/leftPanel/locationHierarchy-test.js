import React from 'react';
import {shallowToJson} from 'enzyme-to-json';
import {shallow, mount} from 'enzyme';
import {IntlProvider} from 'react-intl';

import LocationHierarchy from 'components/admissionLocation/leftPanel/locationHierarchy';
import admissionLocationFunctionsMock from 'components/__mocks__/admissionLocationFunctions-mock';
import messages from 'i18n/messages';

const intlProvider = new IntlProvider({locale: 'en', messages: messages['en']}, {});
const {intl} = intlProvider.getChildContext();
const testData = {
    props: {
        admissionLocationFunctions: admissionLocationFunctionsMock
    },
    context: {
        intl: intl
    }
};

describe('LocationHierarchy', () => {
    it('Renders correctly Location hierarchy', () => {
        const locationHierarchy = shallow(
            <LocationHierarchy isOpen={true} admissionLocationFunctions={testData.props.admissionLocationFunctions} />,
            {context: testData.context}
        );

        expect(shallowToJson(locationHierarchy)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnClickTitle = jest.spyOn(LocationHierarchy.prototype, 'onClickTitle');
        const spyOnClickIcon = jest.spyOn(LocationHierarchy.prototype, 'onClickIcon');
        const spySetState = jest.spyOn(testData.props.admissionLocationFunctions, 'setState');
        const locationHierarchy = mount(
            <LocationHierarchy isOpen={true} admissionLocationFunctions={testData.props.admissionLocationFunctions} />,
            {context: testData.context}
        );

        locationHierarchy.find('li.title span').simulate('click');
        expect(spyOnClickTitle).toHaveBeenCalled();
        expect(spySetState).toHaveBeenCalledWith({
            activeUuid: null,
            isOpen: true,
            activePage: 'listing',
            pageData: {}
        });

        locationHierarchy.find('li.title i').simulate('click');
        expect(spyOnClickIcon).toHaveBeenCalled();
        expect(spySetState).toHaveBeenCalledWith({isOpen: false});
    });

    it('Should function run properly', () => {
        const locationHierarchy = mount(
            <LocationHierarchy isOpen={true} admissionLocationFunctions={testData.props.admissionLocationFunctions} />,
            {context: testData.context}
        );
        const hierarchyFunction = locationHierarchy.instance().hierarchyFunction;
        const spySetActiveLocationUuid = jest.spyOn(testData.props.admissionLocationFunctions, 'setActiveLocationUuid');
        const spySetState = jest.spyOn(testData.props.admissionLocationFunctions, 'setState');
        const spySetAdmissionLocationIsOpen = jest.spyOn(hierarchyFunction, 'setAdmissionLocationIsOpen');

        expect(hierarchyFunction.getActiveUuid()).toBe('baf7bd38-d225-11e4-9c67-080027b662ec');
        expect(hierarchyFunction.getAdmissionLocations()).toBe(
            testData.props.admissionLocationFunctions.getAdmissionLocations()
        );

        hierarchyFunction.setActiveUuid('bb0e512e-d225-11e4-9c67-080027b662ec');
        expect(spySetActiveLocationUuid).toHaveBeenCalledWith('bb0e512e-d225-11e4-9c67-080027b662ec');

        hierarchyFunction.toggleIsOpen('baf7bd38-d225-11e4-9c67-080027b662ec', true);
        expect(spySetAdmissionLocationIsOpen).toHaveBeenCalledWith('baf7bd38-d225-11e4-9c67-080027b662ec', false);

        hierarchyFunction.toggleIsOpen(null, true);
        expect(spySetState).toHaveBeenCalledWith({isOpen: false});

        hierarchyFunction.setAdmissionLocationIsOpen('baf7bd38-d225-11e4-9c67-080027b662ec', true);
        const admissionLocations = hierarchyFunction.getAdmissionLocations();
        expect(admissionLocations['baf7bd38-d225-11e4-9c67-080027b662ec'].isOpen).toBe(true);
    });
});
