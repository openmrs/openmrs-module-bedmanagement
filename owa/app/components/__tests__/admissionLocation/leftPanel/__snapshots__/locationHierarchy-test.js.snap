// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationHierarchy Renders correctly Location hierarchy 1`] = `
<div
  className="left-container"
>
  <ul>
    <li
      className="title"
    >
      <i
        aria-hidden="true"
        className="fa fa-minus-square"
        onClick={[Function]}
      />
      <span
        className=""
        onClick={[Function]}
      >
        Admission Locations
      </span>
    </li>
    <HierarchyItem
      admissionLocation={
        Object {
          "description": "Hospital General Ward Section",
          "isHigherLevel": true,
          "isOpen": false,
          "name": "General Ward",
          "parentAdmissionLocationUuid": "c1e42932-3f10-11e4-adec-0800271c1b75",
          "uuid": "baf7bd38-d225-11e4-9c67-080027b662ec",
        }
      }
      hierarchyFunction={
        Object {
          "getActiveUuid": [Function],
          "getAdmissionLocations": [Function],
          "setActiveUuid": [Function],
          "setAdmissionLocationIsOpen": [Function],
          "setState": [Function],
          "toggleIsOpen": [Function],
        }
      }
      isParentOpen={true}
      key="baf7bd38-d225-11e4-9c67-080027b662ec"
    />
    <HierarchyItem
      admissionLocation={
        Object {
          "description": "Hospital Labour Ward Section",
          "isHigherLevel": true,
          "isOpen": false,
          "name": "Labour Ward",
          "parentAdmissionLocationUuid": null,
          "uuid": "bb0e512e-d225-11e4-9c67-080027b662ec",
        }
      }
      hierarchyFunction={
        Object {
          "getActiveUuid": [Function],
          "getAdmissionLocations": [Function],
          "setActiveUuid": [Function],
          "setAdmissionLocationIsOpen": [Function],
          "setState": [Function],
          "toggleIsOpen": [Function],
        }
      }
      isParentOpen={true}
      key="bb0e512e-d225-11e4-9c67-080027b662ec"
    />
  </ul>
</div>
`;
